### PR TITLE
LaTeXTabulars integration

### LINKED ISSUE #74: Use latex_tabular

### PR DESCRIPTION
**Pull request by abelsiqueira**
Needs #77, and rebased after that.

Closes #74

### CODE REVIEW

File: src/bmark/tables.jl [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+"""
+safe_latex_AbstractString(s :: AbstractString) = replace(s, "_" => "\\_")
+
+"""`safe_latex_AbstractFloat(s)`
+
+For floats. Bypasses `Inf` and `NaN`. Enclose both the mantissa and the
+exponent in `\\(` and `\\)`.
+"""
+function safe_latex_AbstractFloat(s :: AbstractString)
+  strip(s) in ["Inf"; "NaN"] && return s
```
Comment (dpo) [inline src/bmark/tables.jl]: Could we do `\infty`?

File: src/bmark/tables.jl [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+Formats `x` according to its type. For types `Signed`, `AbstractFloat`,
+`AbstractString` and `Symbol`, it uses a predefined formatting string passed to
+`@sprintf` and then the corresponding `safe_latex_<type>` function.
+
+For type `Missing`, it returns "NA".
+"""
+LTXformat
+
+const headers = Dict{Symbol,String}(:name => "Name",
+                                    :objective => "\$f(x)\$",
```
Comment (dpo) [inline src/bmark/tables.jl]: I've been using `\\(` and `\\)` instead of `\$`.

File: src/bmark/tables.jl [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                                    :dual_feas => "\$\\|\\nabla L\\|\$")
+
+"""
+    latex_tabular_results(io, df, kwargs...)
+
+Creates a latex longtable using LaTeXTabulars of a dataframe of results, formatting
+the output for a ready-to-publication table
```
Comment (dpo) [inline src/bmark/tables.jl]: publication-ready

File: src/bmark/tables.jl [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+Inputs:
+- `io::IO`: where to send the table, e.g.:
+
+      open("file.tex", "w") do io
+        latex_tabular_results(io, df)
+      end
+
+- `df::DataFrame`: Dataframe as outputted by `solve_problems` or the dicts returned by
```
Comment (dpo) [inline src/bmark/tables.jl]: outputted -> output

File: src/bmark/tables.jl (hunk 3 above)
Comment (dpo) [reply src/bmark/tables.jl]: also add full stop.

File: src/bmark/tables.jl [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  `bmark_solvers`.
+
+Keyword arguments:
+- `cols::Array{Symbol}`: Which columns of the `df`. Defaults to using all columns;
+- `ignore_missing_cols::Bool`: If `true`, filters out the columns in `cols` that don't
+  exist in the data frame. Useful when creating tables for solvers in a loop where one
+  solver has a column the other doesn't. If `false`, throws `BoundsError` on that
```
Comment (dpo) [inline src/bmark/tables.jl]: on -> in

File: src/bmark/tables.jl [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  situation.
+- `format_over::Dict{Symbol,Function}`: Overrides format for a specific columns, such as
+
+      format_over=Dict(:name => x->@sprintf("\\textbf{%s}", x) |>
+                    safe_latex_AbstractString)`
+- `header_over::Dict{Symbol,String}`: Overrides header names, such as
+  `header_over=Dict(:dual_feas => "\\\$\\\\nabla f(x)\\\$")`, mind the LaTeX naming.
```
Comment (dpo) [inline src/bmark/tables.jl]: `\\(` ... `\\)`

**Comment by abelsiqueira** [comment]
Done

**Pull request merged**